Add host tag info (#3723)
DataDog/documentation: content/tagging/assigning_tags.md

## Configuration Files

> The host tag is [assigned automatically][6] by the Datadog Agent. To customize the host tag, use the Agent configuration file, `datadog.yaml`.

Reviewer: Shouldn't we refer to those as `host tags` since multiple host tags are allowed? I think we should also give a snippet of host tags configuration in the `datadog.yaml` file i.e: 

```
# Set the host's tags (optional)
tags:
  - <KEY>:<VALUE>   
  - env:prod  
  - role:database
```

Reviewer: I think we should also precise that host tags are attached to all metrics, traces, and logs forwarded by a given Datadog Agent allowing for a better correlation between those data types within Datadog.

Author: This part is specific to the hostname (`host` tag). I updated to make that more clear.

## Environment Variables

> When installing the containerized Datadog Agent, host tags can be set using the environment variable `DD_TAGS`. We automatically collect common tags from [Docker][9], [Kubernetes][10], [ECS][11], [Swarm, Mesos, Nomad, and Rancher][9]. To extract even more tags, use the following options:

Reviewer: ```suggestion
When installing the containerized Datadog Agent, set your host tags using the environment variable `DD_TAGS`. The Datadog containerized Agent automatically collect common tags from [Docker][9], [Kubernetes][10], [ECS][11], [Swarm, Mesos, Nomad, and Rancher][9]. To extract even more tags, use the following options:
```
